File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int kthSmallest(TreeNode root, int k) {
        List<Integer> travel = new ArrayList<>();
        dfs(root, k, travel);
        int res = -1;
        if (travel.size() < k) {
            return res;
        }
        Collections.sort(travel);
        return travel.get(k-1);
    }

    public void dfs(TreeNode root, int k, List<Integer> travel) {
        if (root == null) return;
        travel.add(root.val);
        dfs(root.left, k, travel);
        dfs(root.right, k, travel);
    }
}

File: Data Structures & Algorithms/binary-tree-diameter/submission-0.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    diameterOfBinaryTree(root: TreeNode | null): number {
        let diameter = 0

        const height = (node: TreeNode | null): number => {
            if (!node) {
                return 0
            }

            const l = height(node.left)
            const r = height(node.right)

            diameter = Math.max(diameter, l+r)

            return 1+ Math.max(l, r)
        }

        height(root)

        return diameter
    }
}